app/services/db/monthly_sp_auth_count/unique_monthly_auth_counts_by_iaa.rb

```diff
@@ -0,0 +1,194 @@
+module Db
+  module MonthlySpAuthCount
+    module UniqueMonthlyAuthCountsByIaa
+      module_function
+
+      # @param [String] iaa
+      # @param [Symbol] aggregate (one of :sum, :unique, :new_unique)
+      # @return [PG::Result]
+      def call(iaa:, aggregate:)
+        date_range, issuers = iaa_parts(iaa)
+
+        return [] if !date_range || !issuers
+
+        full_months, partial_months = months(date_range).partition { |m| full_month?(m) }
+
+        subquery = [
+          full_month_subquery(issuers: issuers, full_months: full_months),
+          *partial_month_subqueries(issuers: issuers, partial_months: partial_months),
+        ].join(' UNION ALL ')
+
+        select_clause = case aggregate
+        when :sum
+          <<~SQL
+            SUM(billing_month_logs.auth_count)::bigint AS total_auth_count
```

Comment:
>bigint

that's a lot of authentications 😀

Reply:
it was implicitly adding up to be a float, and I figured `bigint` would be safer than something smaller that could get truncated